src/core/thread/dua_manager.cpp

```diff
@@ -342,9 +342,18 @@ void DuaManager::HandleBackboneRouterPrimaryUpdate(BackboneRouter::Leader::State
                                                    const BackboneRouter::Config &aConfig)
 {
     OT_UNUSED_VARIABLE(aConfig);
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    Mle::MleRouter &mle = Get<Mle::MleRouter>();
+#endif
 
     if (aState == BackboneRouter::Leader::kStateAdded || aState == BackboneRouter::Leader::kStateToTriggerRereg)
     {
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+        if (!mle.IsFullThreadDevice() && !mle.GetParent().IsThreadVersion1p1())
+        {
+            return;
```

Comment:
Use `ExitNow()` and `exit` label? (code style guideline require single `return` at the end of meths).

Reply:
Or may be simpler to just use `if` here, something like:
```c++
    if (aState == BackboneRouter::Leader::kStateAdded || aState == BackboneRouter::Leader::kStateToTriggerRereg)
    {
#if OPENTHREAD_CONFIG_DUA_ENABLE
        if (Get<Mle::Mle>().IsFullThreadDevice() || Get<Mle::Mle>().GetParent().IsThreadVersion1p1())
#endif
        {
            UpdateReregistrationDelay();
        }
    }
```

Reply:
Will change it.